src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/UntrackedApprenticeCommitmentsDbContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
+
+namespace SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests
+{
+    public class UntrackedApprenticeCommitmentsDbContext : ApprenticeCommitmentsDbContext
+    {
+        public UntrackedApprenticeCommitmentsDbContext(DbContextOptions<ApprenticeCommitmentsDbContext> options) : base(options)
+        {
+            ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTrackingWithIdentityResolution;
+        }
+
+        public override int SaveChanges()
+        {
+            var r = base.SaveChanges();
+            ChangeTracker.Clear();
+            return r;
+        }
```

Comment:
One concern, what happens if we use SaveChangesAsync()... and the other because we have disabled most tracking in the above line (except for id), how does this ChangeTracker.Clear() affect this?

Reply:
Good point, I'll add the async override too

Reply:
but is it needed at all 

ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTrackingWithIdentityResolution;

is saying no tracking except for Id values

Reply:
I believe that's only for queries.  We also want to avoid tracking entities that we add the to context in the setup phase of tests.  If we don't, then querying those entities for changes in the assert phase simply gives us the references we already have, rather than references to the modified object.

Reply:
This is me getting my head around EF. If QueryTrackingBehavior.NoTrackingWithIdentityResolution wasn't assigned then when we queried the database we would as you say get the object we previously created (this caused me pain), but once assigned when we query the database for that object, it populated that object with the correct values.

I believe this is the behaviour we desire (and therefore I don't understand what value overriding Save.. will give us. 

I might try taking out that code and see if our tests pass. It's the only way for me to learn

Reply:
Just ran tthe code in the following way

1. Without the overrides (worked)
2. Without the NoTrackingWithIdentityResolution (worked)
3. Without the NoTrackingWithIdentityResolution  and overrides (failed)

So, I would assume both implementation do the same thing.